plugin/arcgrid/src/main/java/it/geosolutions/imageio/plugins/arcgrid/raster/AsciiGridRaster.java

```diff
@@ -688,20 +699,50 @@ public WritableRaster readRaster(ImageReadParam param) throws IOException {
 							// number of white spaces smaller than the requested
 							// number. Thus, I need to manually count the
 							// remaining number of spaces.
-
-							imageIS.seek(markedPos.longValue());
-							samplesCounted = (int) key;
+						  samplesToSkip = samplesToThrowAwayBeforeFirstValidSample - key;
+							// this should be getting index values set from
+						  if (LOGGER.getLevel() == Level.FINEST) {
+								LOGGER.finest(String.format("[lookup] Found nearest (%d) indexed stream position %d for sample %d%n",
+							    samplesToThrowAwayBeforeFirstValidSample,
+							    markedPos,
+							    key
+						    ));
+						  }
+
+							indexedSeekPos = markedPos;
+							nextSampleIndex = key;
+							// here we subtract by one, because the index stores the number of samples read at a particular stream position,

```

Comment:
this comment needs to go

Reply:
you have some outstanding comments like this, do you intend to resolve them prior to this PR being merged?

Reply:
Yep, I've been chipping away at this in the evenings, I'll try and clean it up tonight.

Reply:
All crufty comments hopefully gone, plus some slightly opinionated changes to the existing comments to keep it to why rather than what, especially on the bits that had be scratching my head for so long